src/pal/src/file/directory.cpp

```diff
@@ -383,44 +384,42 @@ GetCurrentDirectoryW(
              OUT LPWSTR lpBuffer)
 {
     DWORD dwWideLen = 0;
-    DWORD dwLastError = 0;
+    DWORD dwLastError = ERROR_BAD_PATHNAME;
 
     char  *current_dir;
-    int   dir_len;
 
     PERF_ENTRY(GetCurrentDirectoryW);
     ENTRY("GetCurrentDirectoryW(nBufferLength=%u, lpBuffer=%p)\n",
           nBufferLength, lpBuffer);
 
-    current_dir = PAL__getcwd( NULL, MAX_LONGPATH + 1 );
 
+    current_dir = PAL__getcwd( NULL, 0 );
+    
     if ( !current_dir )
     {
         WARN( "PAL__getcwd returned NULL\n" );
-        dwLastError = DIRGetLastErrorFromErrno();
         goto done;
     }
 
-    dir_len = strlen( current_dir );
```

Comment:
For efficiency, we should keep this. Passing -1 as input length to MultiByteToWideChar causes it to call strlen for you. Since MultiByteToWideChar is called twice, we end up calling strlen twice. Better to just call it once here.